exporters/stats_stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporter.java

```diff
@@ -0,0 +1,277 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.stats.stackdriver;
+
+import static com.google.api.client.util.Preconditions.checkArgument;
+import static com.google.api.client.util.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static io.opencensus.exporter.stats.stackdriver.StackdriverExportUtils.createMetricDescriptor;
+import static io.opencensus.exporter.stats.stackdriver.StackdriverExportUtils.createTimeSeriesList;
+
+import com.google.api.MetricDescriptor;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.cloud.monitoring.v3.MetricServiceSettings;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.monitoring.v3.CreateMetricDescriptorRequest;
+import com.google.monitoring.v3.CreateTimeSeriesRequest;
+import io.opencensus.common.Duration;
+import io.opencensus.stats.Stats;
+import io.opencensus.stats.View;
+import io.opencensus.stats.View.Name;
+import io.opencensus.stats.ViewData;
+import io.opencensus.stats.ViewManager;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * Exporter to Stackdriver Monitoring Client API v3.
+ *
+ * <p>Example of usage on Google Cloud VMs:
+ *
+ * <pre><code>
+ *   public static void main(String[] args) {
+ *     StackdriverExporter exporter = StackdriverExporter.createWithProjectId(
+ *         "MyStackdriverProjectId", Duration.fromMillis(100000));
+ *     exporter.registerView(myView);
+ *     ... // Do work.
+ *   }
+ * </code></pre>
+ */
+public final class StackdriverExporter {
+
+  private final String projectId;
+  private final MetricServiceClient metricServiceClient;
+  private final WorkerThread workerThread;
+
+  private static volatile StackdriverExporter exporter = null;
+
+  private static final Duration ZERO = Duration.create(0, 0);
+
+  @VisibleForTesting
+  StackdriverExporter(
+      String projectId,
+      MetricServiceClient metricServiceClient,
+      Duration exportInterval,
+      ViewManager viewManager) {
+    checkNotNull(exportInterval, "exportInterval");
+    checkArgument(exportInterval.compareTo(ZERO) > 0, "Duration must be positive");
+    this.projectId = checkNotNull(projectId, "projectId");
+    this.metricServiceClient = metricServiceClient;
+    this.workerThread = new WorkerThread(
+        projectId, metricServiceClient, exportInterval, viewManager);
+    workerThread.start();
```

Comment:
It would be better to not start the thread in the constructor, because the "this" reference could escape from the constructor if WorkerThread were ever made non-static.  You could start it in `createInternal` instead.